### PR TITLE
Fix plane_offset() of BlockOffset struct for chroma planes

### PR DESCRIPTION
**Pull request by ycho**
It has been wrong when SB is partitioned, thus we were using a block of codes doing that.

### CODE REVIEW

**Comment by fbossen** [comment]
Looks like a nice cleanup

File: src/context.rs [hunk 1]
```diff
@@ -1167,11 +1167,13 @@ impl BlockOffset {
...
   pub fn plane_offset(&self, plane: &PlaneConfig) -> PlaneOffset {
     let po = self.sb_offset().plane_offset(plane);
+
     let x_offset = self.x & LOCAL_BLOCK_MASK;
     let y_offset = self.y & LOCAL_BLOCK_MASK;
+
     PlaneOffset {
-      x: po.x + (x_offset << BLOCK_TO_PLANE_SHIFT),
-      y: po.y + (y_offset << BLOCK_TO_PLANE_SHIFT)
+        x: po.x + (x_offset >> plane.xdec << BLOCK_TO_PLANE_SHIFT),
```
Comment (smarter) [inline src/context.rs]: Shouldn't this be << (BLOCK_TO_PLANE_SHIFT - plane.xdec), similar to the other plane_offset function ?

**Comment by ycho** [comment]
@lu-zero 
I've revised one of bench function, write_b_bench(), since for ex 8x8 tx shoud be applied
to (bx,by) , where both bx and by are even number, i.e. on 8x8 grid points on 4x4 BlockOffset (i.e. MI) grids.

File: src/context.rs (hunk 1 above)
Comment (ycho) [reply src/context.rs]: @smarter, hey more essence of the reason that I think I missed when I originally replied Aug, 2018 is now here! https://github.com/xiph/rav1e/issues/1062#issuecomment-469812492

**Pull request merged**